src/Header.php

```diff
@@ -628,50 +628,42 @@ private function extractHeaderExtensions(): void {
 
                     // Get all potential extensions
                     $extensions = explode(";", substr($value, $pos + 1));
-                    $previousKey = null;
-                    $previousValue = '';
+                    $extValues = [];
 
                     foreach ($extensions as $extension) {
                         if (($pos = strpos($extension, "=")) !== false) {
                             $key = substr($extension, 0, $pos);
                             $key = trim(rtrim(strtolower($key)));
 
                             $matches = [];
-
                             if (preg_match('/^(?P<key_name>\w+)\*/', $key, $matches) !== 0) {
                                 $key = $matches['key_name'];
-                                $previousKey = $key;
 
                                 $value = substr($extension, $pos + 1);
                                 $value = str_replace('"', "", $value);
-                                $previousValue .= trim(rtrim($value));
-
-                                continue;
-                            }
 
-                            if (
-                                $previousKey !== null
-                                && $previousKey !== $key
-                                && isset($this->attributes[$previousKey]) === false
-                            ) {
-                                $this->set($previousKey, $previousValue);
+                                if (!isset($extValues[$key])) {
+                                    $extValues[$key] = '';
+                                }
+                                $extValues[$key] .= trim(rtrim($value));
```

Comment:
What's the point in trim(rtrim())? `trim` already removes whitespaces from beginning and end.

Reply:
I agree with you, this is an unnecessary part. It's leftover from the original code.